### DISCUSSION TITLE
Can colors be disabled for telnet connections?

### DISCUSSION THREAD

**Discussion by albert-elmore** [evennia/evennia#2448] [Evennia Questions] [answered]
I am working on building a game that can be played on much older machines that are VT52 compatible or worse. So far Evennia is working pretty good out of the box. However I my terminal is not able to filter out color codes. Is there a way to disable colors?

**Comment by kovitikus** [comment]
I believe you are looking for this method here 

https://github.com/evennia/evennia/blob/master/evennia/utils/ansi.py#L482

Well, there's some other strip ANSI stuff in the ANSI parser above this section I linked, but I don't know the specifics on the differences. It says that one is for Evennia specific stuff or something.

If I'm wrong or if there's more you should know about it, I'm certain someone else will chime in. :)

I haven't actually done any such stripping myself.

**Reply by albert-elmore** [reply]
I take it this isn't something I can turn on or add to my settings.py file .. Any advice on how to use or interact with this? 

I've only installed and started using Evennia this evening. So my knowledge is very limited.

**Reply by kovitikus** [reply]
I know there's others in the community who have done this same specific thing you're attempting, so they may be better suited to help you. I've never tried stripping. It's definitely not as simple as a settings entry. But I also doubt it's super difficult either. The method does most of the work for you. You simply have to give it a string.

I believe others have hooked this into the msg path so that everything gets stripped or something.

It's worth it for you to read the documentation on how msg works. https://www.evennia.com/docs/latest/Messagepath.html

Pretty much all of the Developer Central documentation articles are something I recommend you read, and then reread, and then read again. Any free moment you have should be reading through the API and documentation, just to familiarize yourself with it. It's how I gained an understanding of what I was doing, from knowing nothing about Python or Evennia.

**Reply by albert-elmore** [reply]
Good advice. I will drench my self in the docs.

**Comment by Griatch** [comment, marked as answer]
@albertelmore It's _very_ rare for a terminal to not understand ANSI, so when a client does not provide an ANSI TTYPE/MTTS identifier we make the assumption that it supports colors but is just not communicating it correctly. 

There is no need to change the server to just disable colors for yourself though. The `options` command in Evennia will give you the information the client has been sending you. Use 

    options 

to see the info and 

    options/save ANSI=False 

To permanently enforce ansi-colors off no matter what your client tells the server.

Additionally, if the client reports the terminal we could expand the logic server-side. Check if your `options` output contain a `TERM` element - If it does, and it's something uniquely identifiable (like `VT52` etc) we could add a special catch in the telnet protocol of Evennia to automatically turn off colors for clients reporting such a terminal without you needing to do so manually.

**Reply by kovitikus** [reply]
Clearly I was way off. Thanks for the clarification.

**Reply by albert-elmore** [reply]
@Griatch Ok yes! I tried `options/save ANSI=False` and that worked! Thanks! Should this solution persist with my user after I logout and back in? Or is that what you're getting at with regard to adding some logic server-side?

I am on an EPSON PX-8 as a terminal (connected to Linux Pi3 box over RS232). I have yet to figure out which terminal standard the PX-8 conforms to. I desperately want to use this machine as a daily driver. Or at east for MUD and music coding. 

I manually set $TERM on the Pi3 to VT52 which has at least gained me the ability to use the backspace key. However there are many other quirks I need to work on. Screen clearing is another huge limitation I need to find a fix or work around for. No Vim until I can get that figured out :(

Thanks for the quick response and solution! I'll dig into the documentation and come back with more questions when I get stuck.

**Reply by Griatch** [reply]
To answer your question, yes, `options/save` should persist your preference when you log back in later. Of course it can't know that you are you when you first connect to the game though (you must log in first). So for your greeting/login screen you need to either make your own version without color (see `mygame/server/conf/connection_screen.py`) or enter `screenreader` at the login screen to switch to screenreader mode (intended for blind users and thus also lacking colors).
